src/VisualStudio/IntegrationTest/New.IntegrationTests/CSharp/CSharpAddMissingUsingsOnPaste.cs

```diff
@@ -160,9 +161,14 @@ static void Main(string[] args)
 
         private async Task PasteAsync(string text, CancellationToken cancellationToken)
         {
+            var provider = await TestServices.Shell.GetComponentModelServiceAsync<IAsynchronousOperationListenerProvider>(HangMitigatingCancellationToken);
+            var waiter = (IAsynchronousOperationWaiter)provider.GetListener(FeatureAttribute.AddImportsOnPaste);
```

Comment:
note: i don't think we shoudl do this *in* paste.  paste should just send the paste through.  however, the test that wants to validate add-import-on-paste can itself then wait on this async work.

Reply:
All of the tests need to validate this. Even if no work is added the waiter should be there to validate that it's not being triggered right? Otherwise we could be incorrectly testing the negative. We could accidentally have it enabled and not wait on the work being done, so the test would still pass for broken code.

Reply:
> All of the tests need to validate this. Even if no work is added the waiter should be there to validate that it's not being triggered right? Otherwise we could be incorrectly testing the negative. We could accidentally have it enabled and not wait on the work being done, so the test would still pass for broken code.

That's a compelling argument.  in that case, i woudl say that we're not waiting on AddImportsOnPaste, we're effectivley waiting on *any* paste feature.  so i would call this FeatureAttribute.Paste :)

(doesn't ahve to happen now).